src/sql/parts/common/customInputConverter.ts

```diff
@@ -183,6 +188,18 @@ function getNotebookFileExtensions(instantiationService: IInstantiationService):
 	});
 }
 
+/**
+ * Checks whether the given EditorInput is set to either undefined or notebook mode
+ * @param input The EditorInput to check the mode of
+ */
+function hasNotebookFileMode(input: EditorInput): boolean {
```

Comment:
The design of notebooks is such that we can support more than just the ipynb file type. The notebookService now has `getSupportedFileExtensions` method which should be used instead of hardcoding a filename.

Reply:
And note I added the `withService` function to this file so that you can go from the instantiationService -> NotebookService easily. You'll want to use that.